app/controllers/welcome_controller.rb

```diff
@@ -0,0 +1,17 @@
+class WelcomeController < ApplicationController
+  before_action :set_nav
```

Comment:
Neat use of a `before_action`. This is pretty similar in concept to how you'd do this with real users.